### DISCUSSION TITLE
Meeting 2024-07-09

### DISCUSSION THREAD

**Discussion by adamgreig** [rust-embedded/wg#773] [General]
This is the agenda for today's meeting. Please add anything you'd like to discuss below!

* Meetings: Tuesday 8pm Europe/Berlin time
* [Join the Matrix chat](https://matrix.to/#/#rust-embedded:matrix.org)
* [Nominated issues](https://github.com/search?q=org%3Arust-embedded+label%3Anominated+is%3Aopen&type=Issues)
* [IRC/Matrix logs](https://libera.irclog.whitequark.org/rust-embedded/2024-07-09)

### Announcements

* svdtools 0.3.17 released

**Comment by adamgreig** [comment]
From @jamesmunns for the resources team:

We've received a PR for adding commercially available books ("Simplified Embedded Rust" books from apollo labs): https://github.com/rust-embedded/awesome-embedded-rust/pull/453

 * To date, we've not had any "paid only" material on the list
 * We HAVE had "paid adjacent" material, e.g. training material from Ferrous, TweedeGolf, etc., where the MATERIAL was public, but was for commercial purposes
 * IMO the choices are:
    * We codify that we don't accept private commercial items
    * We set up a separate section for paid material, and merge this PR in that section

_Originally posted by @adamgreig in https://github.com/rust-embedded/wg/discussions/772#discussioncomment-9939855_

**Reply by jamesmunns** [reply]
As a note, we decided to go ahead and merge this. We can remove this from this weeks agenda.

**Comment by adamgreig** [comment]
From @jamesmunns for the cortex-m team:

* Poke the cortex-m team to see if it's reasonable to add a way to eagerly detect when we are missing a "real" linker script, e.g. force a linker error for a missing symbol defined in the c-m-rt link.x and referenced by the c-m-rt crate, so people don't end up with "no loadable section" errors from probe-rs.

* I feel like I've seen people asking about that more recently, not sure if anything has changed, but it feels like something we can do something about, at least in the typical case.

_Originally posted by @adamgreig in https://github.com/rust-embedded/wg/discussions/772#discussioncomment-9939865_

**Comment by adamgreig** [comment]
RISC-V fallible functions are almost done: https://github.com/rust-embedded/riscv/pull/222

_Originally posted by @romancardenas in https://github.com/rust-embedded/wg/discussions/772#discussioncomment-9939904_

**Reply by romancardenas** [reply]
- The proposed `result` module can be found [here](https://github.com/rust-embedded/riscv/blob/77b692fd4216ab1f16831cab8bd86b86ea3c27f8/riscv-pac/src/result.rs).
- The module is not part of `riscv`, but `riscv-pac`. However, now `riscv` reexports all the elements in `riscv-pac`, so in essence it can be seen as a module of `riscv`. The idea is to isolate traits and the new `Error` enum from future breaking changes in `riscv`.

**Comment by jamesmunns** [comment]
[RFC3614](https://github.com/rust-lang/rfcs/blob/master/text/3614-project-goals.md) introduced "Project Goals" which are intended to steer the focus and "big picture" of work in the project. The initial batch of goals, "2024H2", have already been selected.

Some external teams, like Rust for Linux, have articulated their desires for the project, and have been included on the ["accepted" list](https://rust-lang.github.io/rust-project-goals/2024h2/slate.html).

As a WG, we could consider enumerating any goals we would like to propose to be added for the next selection process, "2025H1", which will begin discussions in October 2024, and will be selected in December 2024.

I'd like to recommend we consider whether there are any coherent goals we can propose to be part of the next project, to support any stumbling blocks in the embedded ecosystem, or to better position Rust as a preferrable choice for embedded development.

**Reply by ejpcmac** [reply]
Would stabilizing special ABIs (like the ones needed for crossing the S/NS TrustZone-M boundary, RISC-V interrupts or naked functions) be a good candidate for that?

**Reply by jamesmunns** [reply]
@ejpcmac it could be! In general, I think we might want to do something like:

* Start gathering proposals
* Do some curation/prioritization as a working group
* Raise those to the project for tracking

**Comment by boozook** [comment]
I wrote a proposal [issue](https://github.com/rust-lang/cargo/issues/14208) for cargo. I am free right now and can make some patches for this. But before that, I need the opinion of the members of the working group and some advice.

TL;DR: In case of using user's custom target spec json file, in the build-script context there is not enough info about the target available to configure non-rust compilers (e.g. using cc or bindgen). As well as for rustc, using autocfg crate.
In other words, there is not enough existing env vars for build-script-environment.

**Comment by BartMassey** [comment]
Status of Discovery Book triage:

* I'm working in https://github.com/BartMassey-upstream/discovery-mb2 . The `dev` branch should usually contain the latest completed stuff.
* Have restructured as specified.
* Have updated all the code to build cleanly and run with current infrastructure.
* My next task is to go over the text chapter by chapter and clean up.
* @hdoordt is working on a couple of new chapters. He's on a branch of the old book now, but I'll merge his stuff to my repo and consolidate everything shortly.

Open issues:

* Right now the code is mixed in with the book. This works, but makes it awkward for people to do things like grab the code and run it. Worth fixing?

* There's a static globals problem (sigh) with one of the functions in `nrf52833-hal`. It compiles and works right now, but I've suppressed a compiler warning. I'll file an issue.
